src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,152 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
```

Comment:
Probably we do not need explicitly initialise `size` here. Point from **Common Mistakes** was mostly related to cases, when we need to make some non-default initialisation.